pkg/sql/schemachanger/scbuild/builder_state.go

```diff
@@ -395,6 +396,72 @@ func (b *builderState) ComputedColumnExpression(
 	return parsedExpr, newTypeT(typ)
 }
 
+// RetrieveTableWithIndex implements the scbuildstmt.TableHelpers interface.
+func (b *builderState) RetrieveTableWithIndex(
+	ctx context.Context, index *tree.TableIndexName, ifExists bool,
+) (catalog.TableDescriptor, catalog.Index) {
+	// Attempt to resolve the table's prefix, regardless of whether the table name is provided or not.
+	tableName, indexName := index.Table, index.Index
+	resolvedTablePrefix := b.cr.MayResolveObjectNamePrefix(ctx, tableName.ObjectNamePrefix)
```

Comment:
This part seems suspicious. How can this be expected to work if the table name is empty?